### ISSUE TITLE
🐛 BUG: `<></>` is treated like a default slot and can duplicate content on the page

### ISSUE THREAD

**Issue by tony-sull** [snowpackjs/astro#800]
### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

`<></>` is treated as an unnamed `<slot />`, including the usual `<slot>` behavior of throwing out any _default_ content within the slot when the parent passes in `children`

It's will also repeat multiple times for each `<></>` in the template

### Steps to Reproduce

1. Clone https://github.com/tony-sull/repro-astro-duplicate-slots
2. npm start
3. Error! The `h1` in `Child.astro` is replaced with the `children` content meant to go in the `<slot/>`
4. Bonus: Add extra `<></>` in the child component and the content will get duplicated for each one

### Link to Minimal Reproducible Example (Optional)

https://github.com/tony-sull/repro-astro-duplicate-slots

**Comment by kaleidawave** [comment]
Any progress or workarounds for this?

Possibly related to this but for:

```astro
---
const show_x = true;
---

<div>
    <div>
        {
            show_x ? <>
                <h1>Fragment Content</h1>
            </> :
            <h3>Other Content</h3>
        }
    </div>
    <div>
        <slot/>
    </div>
</div>
```

and

```astro
<Component>
     <h1>Slot content</h1>
</Component>
```

I get:

```html
<div><div>[object Object]<h1>Fragment Content</h1></div><div><h1 class="astro-ldKR6c5M">Slot content</h1></div></div>
```

Not sure what that `[object Object]` is doing there 😆

**Comment by jasikpark** [comment]
probably a naive `.toString()` on something?

**Comment by johnste** [comment]
I'm  also seeing the `[object Object]` output as well.  When using fragments in child components (i.e. not in the pages/index.astro one, but one that I import). It appears that the issue started with 0.18 release.

Been trying to get a good repro but codesandbox isn't working well for me.

**Comment by jonathantneal** [comment]
Here is a reproduction (from @johnste) https://github.com/johnste/astro-fragment-issue

**Comment by tony-sull** [comment]
There's some extra logic around slots like supporting named slots, but yeah they are treated pretty similarly.

It is probably just a parser fix that's needed, but on first pass I wasn't sure exactly why the existing logic is making the checks it does and would need to really dig into it to find a proper fix.

I'm actually going to close this one since it isn't going to be an issue with the new compiler.  In the meantime, anyone that hits this can avoid the bug by using `<Fragment>` instead of the `<>` shorthand 👍

**Comment by jasikpark** [comment]
I would say it makes more sense to leave it open and label it as conpiler-v2

**Comment by jasikpark** [comment]
I believe `compiler-v2` labels are in the "blocked" section because they are accepted bugs that are waiting for the release of the v2 compiler

**Comment by tony-sull** [comment]
Just moved it, thanks!  I was on mobile last night when I reopened the bug and GitHub's modal for the Projects board went crazy, it opened hanging off the left edge of the screen then changed the selection 5 or 6 times before I got the tab closed :confused:

**Comment by jasikpark** [comment]
Yeah, I'm on the mobile app rn, and I can only add issues to a project, not so much change their status 😅

**Comment by natemoo-re** [comment]
Confirming that this _should be fixed_ in the `astro@next` preview, but we still have some bugs related to `<></>` support. In general, Fragments should no longer be needed but we'll support them for backwards compatability.

See https://github.com/snowpackjs/astro-compiler-next/pull/126

**Comment by jasikpark** [comment]
Oops didn't see your stackblitz link, but https://codesandbox.io/s/runtime-violet-g4c28?file=/src/components/Test.astro works too!

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
